libbeat/common/match/optimize.go

```diff
@@ -0,0 +1,204 @@
+package match
```

Comment:
There are kind of two solutions here:

* We fix it automatically for the user what this code is doing
* We log a warning to the that his regexp is not optimal and that he could optimize it when doing x
* Doing a combination of both

I would opt for version 3. But will any user ever have a look at it an optimize the regexp query if we tell him it is not optimal? Main thing about doing 1 is that this could introduce potential errors. Means the regexp we use internally is not the regexp someone has in the config file. So it would be good to at least have some debug message with the now used regexp.